packages/serverless-orchestration/src/ServerlessHub.js

```diff
@@ -73,13 +81,17 @@ hub.post("/", async (req, res) => {
     // in the next iteration when the hub is called again.
     await _saveQueriedBlockNumber(req.body.configFile, latestBlockNumber);
 
-    // Loop over all config objects in the config file and for each append a call promise to the promiseArray.
+    // Loop over all config objects in the config file and for each append a call promise to the promiseArray. Note
+    // that each promise is a race between the serverlessSpoke command and a `_rejectAfterDelay`. This places an upper
+    // bound on how long each spoke can take to respond, acting as a timeout for each spoke call.
     let promiseArray = [];
     let botConfigs = {};
     for (const botName in configObject) {
       const botConfig = _appendEnvVars(configObject[botName], botName, lastQueriedBlockNumber, latestBlockNumber);
       botConfigs[botName] = botConfig;
-      promiseArray.push(_executeServerlessSpoke(spokeUrl, botConfig));
+      promiseArray.push(
+        Promise.race([_executeServerlessSpoke(spokeUrl, botConfig), _rejectAfterDelay(hubConfig.rejectSpokeDelay)])
```

Comment:
this is the key change: each spoke call is a race between the spoke execution and a rejecting delay. whichever ends first wins. i.e if the reject ends first, the call timed out.

Reply:
Why was this an issue before without the race method?

Reply:
Before this, the spokes could take unlimited time to respond. This race places an upper bound on how long a spoke can take to reply.